### PR TITLE
Learning Arithmetic Operators

### PR DESCRIPTION
**Pull request by Zain-Bajwa**
# Dscription
Testing description a,b c.

### CODE REVIEW

File: src/data_types/test_dictionaries.py [hunk 1]
```diff
@@ -77,3 +77,6 @@ def test_dictionary():
     assert dictionary_for_string_keys['sape'] == 4139
     assert dictionary_for_string_keys['guido'] == 4127
     assert dictionary_for_string_keys['jack'] == 4098
+
+
+test_dictionary()
```
Comment (arslanashraf7) [inline src/data_types/test_dictionaries.py:82]: Missing a new line here and on other files too. Syntactically the python files or any other files it's reasonable to add an empty new line at the end of the files.

File: src/operators/test_identity.py [hunk 2]
```diff
@@ -5,31 +5,50 @@
...
+import copy
 
 def test_identity_operators():
     """Identity operators"""
 
     # Let's illustrate identity operators based on the following lists.
     first_fruits_list = ["apple", "banana"]
     second_fruits_list = ["apple", "banana"]
-    third_fruits_list = first_fruits_list
+    third_fruits_list = first_fruits_list   #Shallow copy
```
Comment (arslanashraf7) [inline src/operators/test_identity.py]: Some things that are purely technically related to how the language handles it don't need comments.

File: src/operators/test_identity.py [hunk 3]
```diff
@@ -5,31 +5,50 @@
...
+
+
+    third_fruits_list[0] = "Mango"
+    print(first_fruits_list)
+
+    assert first_fruits_list is third_fruits_list
+    assert first_fruits_list == third_fruits_list
+
+
+    third_fruits_list = copy.deepcopy(first_fruits_list)   #Deep copy copy
```
Comment (arslanashraf7) [inline src/operators/test_identity.py]: Same as above

File: src/operators/test_identity.py [hunk 4]
```diff
@@ -5,31 +5,50 @@
...
+    # third_fruits_list[0] = "Mango"
```
Comment (arslanashraf7) [inline src/operators/test_identity.py]: Could you remove this comment

File: src/operators/test_identity.py [hunk 5]
```diff
@@ -5,31 +5,50 @@
...
+    assert first_fruits_list == third_fruits_list
 
     # is not
     # Returns true if both variables are not the same object.
 
     # Example:
     # first_fruits_list and second_fruits_list are not the same objects, even if they have
     # the same content
-    assert first_fruits_list is not second_fruits_list
+    # assertsert first_fruits_list is not second_fruits_list
```
Comment (arslanashraf7) [inline src/operators/test_identity.py]: Why did we comment this?

File: src/operators/test_identity.py [hunk 6]
```diff
@@ -5,31 +5,50 @@
...
 
     # To demonstrate the difference between "is" and "==": this comparison returns True because
     # first_fruits_list is equal to second_fruits_list.
-    assert first_fruits_list == second_fruits_list
+    # assert first_fruits_list == second_fruits_list
```
Comment (arslanashraf7) [inline src/operators/test_identity.py]: Any specific reason for commenting this assert? Shouldn't it test something?

**Review by arslanashraf7 — CHANGES_REQUESTED** [review]
I did the first pass on your PR.

Can you add more examples of your own tests? Try to not comment on the assert statements and add more tests with what you practiced?

File: src/operators/test_identity.py (hunk 5 above)
Comment (Zain-Bajwa) [reply src/operators/test_identity.py]: I created a deep copy for practice. In this scenario, this assert was creating an error. So I did comment this line. Yes It's my mistake either I should have corrected it or removed it. Now I have corrected it.

File: src/operators/test_identity.py (hunk 4 above)
Comment (Zain-Bajwa) [reply src/operators/test_identity.py]: Yes, I can remove it. And now I have uncommented it.

File: src/data_types/test_dictionaries.py (hunk 1 above)
Comment (arslanashraf7) [reply src/data_types/test_dictionaries.py:82]: This still has not empty new line at the end of file.

File: src/operators/test_assigment.py [hunk 7]
```diff
@@ -84,11 +84,17 @@ def test_augmented_assignment_operators():
     assert number == 6  # 0b0110
 
     # Assignment: >>=
-    number = 5
+    number = 11
+    print(number >> 2)
     number >>= 3
-    assert number == 0  # (((5 // 2) // 2) // 2)
```
Comment (arslanashraf7) [inline src/operators/test_assigment.py:89]: You have to adjust the assert statement as per the value you changed above for testing purpose which is 11 now.

**Review by arslanashraf7 — CHANGES_REQUESTED** [review]
I added some more comments, can you also practice `*, /` operators in this PR.

In addition, can you change the way you are practicing? for example without commenting and changing the values of the existing tests you should write your own practice changes in the file and write assert statement for that without editing the existing variable values or their asserts?

**Comment by arslanashraf7** [comment]
It is generally a good practice to add brief descriptions of the PR mentioning what the PR does, how can the reviewer test it, and any specific functionality to carefully look around while testing.
For example, in this case, it could be:

#### Description
Adds practice samples for operators (You can even specify which operators). Adjusts the relevant assert statements to pass with your practice samples.

### How to test?
- A look through the code
- Run the tests with pytest
- run the tests and they should all pass.

**Hint:** A bonus in this is look through how you can create a PR template in git so that the headings or the common things automatically populate in the PR description whenever you create a PR for a specific repository.

**Pull request closed without merging**